### PR TITLE
Desktop NTP: prevent image bg loading observer getting destroyed and not getting onload event

### PR DESCRIPTION
**Pull request by petemill**
Also adds debug logging to further diagnose the issue in production builds should the issue persist

Resolves https://github.com/brave/brave-browser/issues/28493

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_new_tab_ui/containers/newTab/index.tsx [hunk 1]
```diff
@@ -217,19 +219,26 @@ class NewTabPage extends React.Component<Props, State> {
...
   trackCachedImage () {
+    console.debug('trackCachedImage')
     if (this.state.backgroundHasLoaded) {
+      console.debug('Resetting to new image')
       this.setState({ backgroundHasLoaded: false })
     }
     if (this.imageSource) {
       const imgCache = new Image()
+      // Store Image in class so it doesn't go out of scope
+      this.imgCache = imgCache
```
Comment (fallaciousreasoning) [inline components/brave_new_tab_ui/containers/newTab/index.tsx:230]: Wow, does this fix it? That's gnarly :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
Good find Pete!

**Comment by simonhong** [comment]
Tested this with my linux local build. It was reproduced very frequently on my linux local build.
and it seems fixed with this PR! 👍🏼

**Review by simonhong — APPROVED** [review]
👍🏼

File: components/brave_new_tab_ui/containers/newTab/index.tsx (hunk 1 above)
Comment (petemill) [reply components/brave_new_tab_ui/containers/newTab/index.tsx:230]: Yep. I had a feeling after I added logging to onload (and onerror) and just was never getting the event called. So I assume it was just garbage collected, or perhaps an optimization on Image element fetching to cancel if it's out of scope entirely. Not too sure which one, but yes it seems to fix it!

**Comment by petemill** [comment]
I left the console.debug entries in for now

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.51.14 Chromium: 111.0.5563.50 (Official Build) nightly (64-bit)
-- | --
Revision | 39f3dfc5023fe60997adc4f6cf5439cfa3e6968a-refs/branch-heads/5563@{#903}
OS | Windows 11 Version 22H2 (Build 22621.1265)
```

I couldn't reproduce the original issue using `1.50.54 Chromium: 110.0.5481.104` but ensured that using the latest version was working as expecting. Uploaded several images varying from ~2-3MBs and ensured it loaded without issues. Used the STR/Cases outlined via https://github.com/brave/brave-browser/issues/28493#issue-1583316000.

**Pull request merged**